api/users.rb

```diff
@@ -79,8 +79,14 @@
 end
 
 post "#{APIPREFIX}/users/:user_id/retire" do |user_id|
-  return {}.to_json if not params["retired_username"]
-  user = User.find_by!(external_id: user_id)
+  if not params["retired_username"]
+    error 500, {message: "Missing retired_username param."}.to_json
+  end
+  begin
+    user = User.find_by(external_id: user_id)
```

Comment:
That confused me as well! But here's the full text of the error:
```
Mongoid::Errors::DocumentNotFound at /api/v1/users/6/retire

message: Document not found for class User with attributes {:external_id=>"6"}. summary: When calling User.find_by with a hash of attributes, all attributes provided must match a document in the database or this error will be raised. resolution: Search for attributes that are in the database or set the Mongoid.raise_not_found_error configuration option to false, which will cause a nil to be returned instead of raising this error.
```
So it appears it's working as the docs say with `find_by()` vs. `find_by!()` - but there's a separate Mongoid exception being thrown.

I'm going to change this code to catch the specific Mongoid exception - it's just better code.

Reply:
:+1: 

So if I understand correctly, the Mongoid.raise_not_found_error option is set to true in our case, so both versions of find_by would raise an exception.

Reply:
Correct - our mongoid.yml:
https://github.com/edx/cs_comments_service/blob/master/config/mongoid.yml
doesn't include the setting and it defaults to true:
https://mongoid.github.io/old/en/mongoid/v3/installation.html

Reply:
Thank you for the clarification.